### ISSUE TITLE
Entity based icons should all support `state_color`

### ISSUE THREAD

**Issue by dermotduffy** [dermotduffy/frigate-hass-card#1793]

**Comment by felipecrs** [comment]
I don't think this is limited to `select` submenu. I think this applies to everywhere entity icons are used, including the camera selector for example.

**Comment by dermotduffy** [comment]
Note: This also won't fix your problem if the goal is to get the icon to white, when state_color is false it still renders it that default "blue" color (for a light, it is that blue when off and bright yellow when on).

I think the fix for you is simply:

```
style:
  color: white
```

... which doesn't yet work, but working on that fix.

**Comment by felipecrs** [comment]
That's why I also "proposed" this:

"Perhaps there's a way to detect when the default color is returned by HA, and then you can patch it to be the previous default color (white)?"

If this were possible, I believe it would be the simplest way to "resolve" this. I.e. restore the previous icon coloring that made more sense for the card as is, yet resolving the missing icons issue.

But I'm not sure if it's possible.

**Comment by felipecrs** [comment]
Then maybe this "patch" can be removed in a future card version, better prepared to handle HA native colors ([hint](https://github.com/dermotduffy/frigate-hass-card/issues/1704#issuecomment-2560449267)).

**Comment by dermotduffy** [comment]
There's no way (beyond a terrible hack) to "detect" the ultimate rendered color. I've submitted the PR to let it the color be explicitly overridden. I'm thinking that, combined:

 - With support for 'state_color' for over entity based menu items.
 - Better menu coloring to match HA `light move`

And perhaps that's good enough for now.

**Comment by felipecrs** [comment]
Yes, I agree, that sounds good. :)

**Comment by haforme** [comment]
I'm thinking that this case is related to a problem that came up when I upgraded to 6.1.2 from 6.0.13.  When I upgraded the entity icon colors stopped working.  An example piece of code is listed below that was working but is not anymore.  I can provide the Frigate Card diagnostics if desired. Thank-you.

  - type: custom:frigate-card-menu-state-icon
    entity: switch.barn_recordings
    title: Record On/Off
    icon: mdi:record-rec
    tap_action:
      action: perform-action
      data:
        entity_id: switch.barn_recordings
      perform_action: switch.toggle

**Comment by felipecrs** [comment]
Do you mean the state colors?

**Comment by felipecrs** [comment]
Although the colors changed, the state colors are still working for me:

![chrome_Zb1EHQbMHd](https://github.com/user-attachments/assets/72630239-60d8-4e7e-a4ac-df85925298bc)

**Comment by haforme** [comment]
Yes, The state colors for entities. (entity_id: switch.barn_recordings)

**Comment by felipecrs** [comment]
I tested with a switch entity here too and it goes yellow normally.

**Comment by dermotduffy** [comment]
@haforme State colors may have changed, but should still work (and certainly do for me). If this is not what you're seeing, please open a new issue and lets discuss.

**Comment by WhistleMaster** [comment]
I have the same issue. The state color is working if no custom icon is specified, otherwise, the icon state does not change.

The following is not working:
```
elements:
  - type: custom:frigate-card-menu-state-icon
    entity: switch.my_switch
    icon: mdi:coach-lamp
    tap_action:
      action: toggle
```

The following is working:
```
elements:
  - type: custom:frigate-card-menu-state-icon
    entity: switch.my_switch
    tap_action:
      action: toggle
```
